src/main/java/org/corfudb/runtime/view/LayoutView.java

```diff
@@ -55,11 +55,14 @@ public int getQuorumNumber() {
     @SuppressWarnings("unchecked")
     public void updateLayout(Layout layout, long rank)
             throws QuorumUnreachableException, OutrankedException, WrongEpochException {
+        runtime.invalidateLayout(); // Force our runtime to adopt latest epoch prior to updating
```

Comment:
well, it is a wrong fix to a non problem.

a client issues updateLayout within its known universe. You do not want to change the universe and maintain the suggested layout.

you also do not want to add a phase to Paxos consensus.

throwing an exception here is the correct behavior, so the model is wrong.

this PR is already closed, no reason to open. we are trying to move forward -- please apply all of the required changes rebased over #244.